test/automation/src/positron/positronEditor.ts

```diff
@@ -22,4 +23,36 @@ export class PositronEditor {
 		await lineLocator.press(press);
 	}
 
+	// This function returns the top value of the style attribute for the
+	// editor's current-line div parent (the current-line div is a child of the div that
+	// has the top attribute). It retries up to 10 times to get the value.
+	// Retries are necessary because the value is not always immediately
+	// available (presumably due to the change from one current line to another).
+	async getCurrentLineTop(): Promise<number> {
+		let retries = 10;
+		let topValue: number = NaN;
+
+		// Note that this retry loop does not sleep.  This is because the
+		// test that uses this function needs to run as quickly as possible
+		// in order to verify that the editor is not executing code out of order.
+		while (retries > 0) {
+			const currentLine = this.code.driver.getLocator(CURRENT_LINE);
+			// Note that '..' is used to get the parent of the current-line div
+			const currentLineParent = currentLine.locator('..');
```

Comment:
These two lines are static and wont change each time they're calculated. The locator is just a reference for how to find something and isn't evaluated until something is done with them. So this can be brought out of the while loop.

Reply:
Good point.  Moved out of the loop.